tests/functional/n/nan_comparison.py

```diff
@@ -0,0 +1,21 @@
+# pylint: disable=missing-docstring, invalid-name, misplaced-comparison-constant,literal-comparison, comparison-with-itself, import-error
+"""Test detection of NaN value comparison."""
+
+import numpy
+x = 42
+a = x is numpy.NaN
+b = x == numpy.NaN  # [nan-comparison]
+c = x == float('nan')  # [nan-comparison]
```

Comment:
I am bit confused here.IMHO the only clean way to test if a float value is nan is to use the `isnan` function of the `math` module. For example:
```
x = float('nan')

x == float('nan')
False

x is float('nan')
False

import math

math.isnan(x)
True

x is math.nan
False
```

Reply:
Okay i take your advice. when the operator is one of 'is' 'is not' '==' '!=' i would suggest to change it to math.isnan(<>)